### PR TITLE
Hide newsfeed description

### LINKED ISSUE #2517: Hide newsfeed description setting ignored
## I found a bug in MagicMirror

I'm using the latest master version on a Raspberri Pi. After the upgrade the showDescription setting is ignored for the newsfeed module.

### PR DESCRIPTION
**Pull request by oemel09**
Fixes #2517

showDescription setting for newsfeed module is checked and applied again.

### CODE REVIEW

**Comment by rejas** [comment]
Looks good. Do you feel brave enough to add a test case for this too?

File: modules/default/newsfeed/newsfeed.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ Module.register("newsfeed", {
 			sourceTitle: item.sourceTitle,
 			publishDate: moment(new Date(item.pubdate)).fromNow(),
 			title: item.title,
-			description: item.description
+			description: this.isShowingDescription ? item.description : null
```
Comment (rejas) [inline modules/default/newsfeed/newsfeed.js:138]: On second though: Shouldnt this be rather checked in the njk-template like the other options too?

**Review by rejas — CHANGES_REQUESTED** [review]

File: modules/default/newsfeed/newsfeed.js (hunk 1 above)
Comment (MichMich) [reply modules/default/newsfeed/newsfeed.js:138]: Agree with @rejas.

Comment (oemel09) [reply modules/default/newsfeed/newsfeed.js]: Changed it, definitely better. Didn't notice the config being passed into the template as well...

File: tests/e2e/modules/newsfeed_spec.js [hunk 2]
```diff
@@ -36,6 +36,10 @@ describe("Newsfeed module", function () {
 		it("should show the newsfeed article", function () {
 			return app.client.waitUntilTextExists(".newsfeed .newsfeed-title", "QPanel", 10000);
 		});
+
+		it("should show the newsfeed description", function () {
+			return app.client.waitUntilTextExists(".newsfeed .newsfeed-desc", "Para instalar esta nueva versión", 10000);
+		});
```
Comment (rejas) [inline tests/e2e/modules/newsfeed_spec.js:42]: And maybe a test where the description is turned off *prettyplease* :-)

File: modules/default/newsfeed/newsfeed.njk [hunk 3]
```diff
@@ -13,13 +13,15 @@
...
             {{ title }}
         </div>
-        <div class="newsfeed-desc small light{{ ' no-wrap' if not config.wrapDescription }}">
-            {% if config.truncDescription %}
-                {{ description | truncate(config.lengthDescription) }}
-            {% else %}
-                {{ description }}
-            {% endif %}
-        </div>
+        {% if (config.showDescription) %}
```
Comment (rejas) [inline modules/default/newsfeed/newsfeed.njk:16]: I am wondering whats occuring more often "if (<condition>)" or "if <condition>" ?

Comment (MichMich) [reply modules/default/newsfeed/newsfeed.njk:16]: I'd say we strip out all unnecessary chars. But that isn't specifically necessary for this PR.

File: tests/e2e/modules/newsfeed_spec.js (hunk 2 above)
Comment (oemel09) [reply tests/e2e/modules/newsfeed_spec.js:42]: Done. Although I'm not sure if this is the right way to test for the absence of a text

Comment (rejas) [reply tests/e2e/modules/newsfeed_spec.js:42]: Mhhh, maybe wait till the title exists and then check if the description is there? Might also be nice to do in the other test

Comment (oemel09) [reply tests/e2e/modules/newsfeed_spec.js:42]: I'm not too familiar with these tests. I tried but couldn't get that to work...

**Comment by rejas** [comment]
I think the test should look like this:


```
it("should NOT show the newsfeed description", async () => {
			await app.client.waitUntilTextExists(".newsfeed .newsfeed-title", "QPanel", 10000);
			const events = await app.client.$$(".newsfeed .newsfeed-desc");
			return expect(events.length).equals(0);
		});
```

**Review by rejas — APPROVED** [review]

**Comment by rejas** [comment]
Nice! thanks again for the PR!

**Comment by MichMich** [comment]
Great PR. Thanks!

**Pull request merged**